docs/sources/flow/reference/components/loki.source.kubernetes.md

```diff
@@ -159,6 +161,19 @@ arguments.
 
 [using clustering]: {{< relref "../../concepts/clustering.md" >}}
 
+### default_instance_label block
+
+You can use the `default_instance_label` block to explicitly define the default "instance" label assigned to the logs 
+scraped from the targets.
+If you don't provide the `default_instance_label` block, the target's address is used as the default instance label.
+
+The following arguments are supported:
+
+Name                     | Type                | Description                                                   | Default | Required
+------------------------ | ------------------- | ------------------------------------------------------------- | ------- | --------
+`enabled`                | `bool`              | Enable setting default instance label value.                | true    | no
```

Comment:
Why do we need a separate enabled value? Would having a value inherently make it true?

Reply:
AFAIK in both Prometheus and Loki, an empty value label is considered equivalent to a label that doesn't exist.

So in this case, you wouldn't really be able to have an empty instance label. I'd agree this makes sense as an attribute instead of a block. It should only override `defaultInstance` when it's expliticly supplied as a non-empty string.

Reply:
Very cool knowing the practice put in place by other parts of the ecosystem :D

I'd try implementing it that way.

Reply:
I just realized that the original requester would like to disable the behavior of setting default `instance` label too 🤦 

And having an empty string to disable the behavior would be breaking change, unless we choose a token to signify the act of disabling, like `defaultInstanceLabel="disabled"` or something, which doesn't seem right to me 🤔 

To have both the ability to disable the behavior and to set a custom default `instance` label, I think we gotta resort to block now (unless having a separate attribute :D)